addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3045,12 +3057,24 @@ const SnippetOptionWidget = Widget.extend({
         // (otherwise we suppose that we should use the actual related color).
         if (params.colorNames && params.colorPrefix) {
             const classes = weUtils.computeColorClasses(params.colorNames, params.colorPrefix);
-            this.$target[0].classList.remove(...classes);
+            const isColorCombination = weUtils.isColorCombinationName(widgetValue);
+            if (params.colorThemeCombination) {
+                if (isColorCombination) {
+                    this.$target[0].classList.remove(...classes.filter((c)=>c.startsWith('o_cc')));
+                } else {
+                    this.$target[0].classList.remove(...classes.filter((c)=>!c.startsWith('o_cc')));
+                }
```

Comment:
- Missing spaces around the `=>` operator.
- But can't this be coded the way we talked about? Since the preset color classes are now supposed to be possible to add alonside other color classes, should not they be moved out of `selectStyle` entirely? So one method for preset, one for bg color and one for bg gradient? And then you can have 3 distinct `_computeWidgetState` call for the same widget? Otherwise, I fear the way you will select the 3 distinct values in the colorpicker will be wrong.